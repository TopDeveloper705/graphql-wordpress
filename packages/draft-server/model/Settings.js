import DataLoader from 'dataloader';
import findByIds from 'mongo-find-by-ids';

export default class Settings {
  constructor(context) {
    this.context = context;
    this.collection = context.db.collection('settings');
    this.loader = new DataLoader(ids => findByIds(this.collection, ids));
  }

  findOneById(id) {
    return this.loader.load(id);
  }

  async updateById(id, doc) {
    const docToUpdate = Object.assign({}, doc);
    const ret = await this.collection.update(
      { _id: id },
      {
        $set: Object.assign({}, docToUpdate, {
          updatedAt: Date.now(),
        }),
      }
    );
    this.loader.clear(id);
    return ret;
  }
}
