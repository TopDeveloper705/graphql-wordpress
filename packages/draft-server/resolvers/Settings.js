const resolvers = {
  Settings: {
    __resolveType(settings) {
      if (settings._id === 'site') {
        return 'SiteSettings';
      } else if (settings._id === 'social') {
        return 'SocialSettings';
      }
      return null;
    },
  },
  SiteSettings: {
    id(settings) {
      return settings._id;
    },
  },
  SocialSettings: {
    id(settings) {
      return settings._id;
    },
  },
  Query: {
    settings(root, { id }, { Settings }) {
      return Settings.findOneById(id);
    },
  },
  Mutation: {
    async updateSiteSettings(root, { id, input }, { Settings }) {
      await Settings.updateById(id, input);
      return Settings.findOneById(id);
    },
    async updateSocialSettings(root, { id, input }, { Settings }) {
      await Settings.updateById(id, input);
      return Settings.findOneById(id);
    },
  },
};

export default resolvers;
