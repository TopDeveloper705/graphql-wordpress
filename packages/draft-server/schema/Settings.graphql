interface Settings {
  id: String
}

type SiteSettings implements Settings {
  id: String
  siteTitle: String
  tagline: String
}

extend type Query {
  settings(id: String): Settings
}

input UpdateSettingsInput {
  siteTitle: String
  tagline: String
}

extend type Mutation {
  updateSettings(id: String!, input: UpdateSettingsInput!): Settings
}
