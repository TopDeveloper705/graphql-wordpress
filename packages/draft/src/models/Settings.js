import Model from './Model';

export default class Settings extends Model {
  constructor(context) {
    super(context);

    this.collection = context.db.collection('settings');
  }
}
