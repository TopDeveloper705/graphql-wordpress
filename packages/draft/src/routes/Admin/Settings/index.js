import React, { Component, Fragment } from 'react';
import { compose, graphql } from 'react-apollo';
import gql from 'graphql-tag';
import Loading from 'components/Loading';
import Form from '../Form';
import { Heading } from '../styled';

/* eslint-disable react/prop-types */

const settingsFields = [
  { label: 'Site Title', prop: 'siteTitle', editable: true },
  { label: 'Tagline', prop: 'tagline', editable: true },
];

@compose(
  graphql(
    gql`
      query SettingsAdminQuery($id: String) {
        settings(id: $id) {
          id
          ... on SiteSettings {
            siteTitle
            tagline
          }
        }
      }
    `,
    {
      options: {
        variables: { id: 'site' },
      },
    }
  ),
  graphql(gql`
    mutation UpdateSettingsMutation($id: String!, $input: UpdateSettingsInput!) {
      updateSettings(id: $id, input: $input) {
        id
        ... on SiteSettings {
          siteTitle
          tagline
        }
      }
    }
  `)
)
export default class SettingsRoute extends Component {
  onSubmit = (e, updates) => {
    e.preventDefault();

    const { settings } = this.props.data;
    this.props.mutate({
      variables: {
        id: settings.id,
        input: updates,
      },
    });
  };

  render() {
    const { data: { loading, settings } } = this.props;

    return (
      <Fragment>
        <Heading>Edit Settings</Heading>
        {loading && !settings ? (
          <Loading />
        ) : (
          <Form
            fields={settingsFields}
            data={settings}
            buttonLabel="Update Settings"
            onSubmit={this.onSubmit}
          />
        )}
      </Fragment>
    );
  }
}
