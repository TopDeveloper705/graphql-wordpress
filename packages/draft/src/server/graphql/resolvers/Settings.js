const resolvers = {
  Settings: {
    __resolveType(settings) {
      if (settings._id === 'site') {
        return 'SiteSettings';
      } else if (settings._id === 'social') {
        return 'SocialSettings';
      } else if (settings._id === 'dashboard') {
        return 'DashboardSettings';
      } else if (settings._id === 'media') {
        return 'MediaSettings';
      }
      return null;
    },
  },
  SiteSettings: {
    id(settings) {
      return settings._id;
    },
  },
  DashboardSettings: {
    id(settings) {
      return settings._id;
    },
  },
  SocialSettings: {
    id(settings) {
      return settings._id;
    },
  },
  MediaSettings: {
    id(settings) {
      return settings._id;
    },
  },
  Query: {
    settings(root, { id }, { Settings }) {
      return Settings.findOneById(id);
    },
  },
  Mutation: {
    async updateSiteSettings(root, { id, input }, { Settings }) {
      await Settings.updateById(id, input);
      return Settings.findOneById(id);
    },
    async updateDashboardSettings(root, { id, input }, { Settings }) {
      await Settings.updateById(id, input);
      return Settings.findOneById(id);
    },
    async updateSocialSettings(root, { id, input }, { Settings }) {
      await Settings.updateById(id, input);
      return Settings.findOneById(id);
    },
    async updateMediaSettings(root, { id, input }, { Settings }) {
      await Settings.updateById(id, input);
      return Settings.findOneById(id);
    },
  },
};

export default resolvers;
