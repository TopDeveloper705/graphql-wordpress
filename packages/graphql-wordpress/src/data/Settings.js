import { toGlobalId } from 'graphql-relay';

class Settings {
  getID() {
    return toGlobalId(this.constructor.name, 'me');
  }

  static getEndpoint() {
    return process.env.WP_SETTINGS_ENDPOINT || 'graphql/v1/settings';
  }
}

export default Settings;
