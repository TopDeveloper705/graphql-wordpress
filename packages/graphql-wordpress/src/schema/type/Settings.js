import { GraphQLObjectType, GraphQLString } from 'graphql';

const SettingsType = new GraphQLObjectType({
  name: 'Settings',
  description: '.',
  fields: {
    title: {
      type: GraphQLString,
      description: 'Site title.',
    },
    description: {
      type: GraphQLString,
      description: 'Site tagline.',
    },
    timezone: {
      type: GraphQLString,
      description: 'A city in the same timezone as you.',
    },
    language: {
      type: GraphQLString,
      description: 'Site locale code.',
    },
  },
});

export default SettingsType;
