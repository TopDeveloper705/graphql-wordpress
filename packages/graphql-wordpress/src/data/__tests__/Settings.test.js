import Settings from 'data/Settings';

const backupEnv = Object.assign({}, process.env);

afterAll(() => {
  process.env = backupEnv;
});

describe('Test Settings data access', () => {
  test('Get endpoint', () => {
    expect(Settings.getEndpoint()).toMatchSnapshot();
  });

  test('Get backup endpoint', () => {
    delete process.env.WP_SETTINGS_ENDPOINT;
    expect(Settings.getEndpoint()).toMatchSnapshot();
  });
});
