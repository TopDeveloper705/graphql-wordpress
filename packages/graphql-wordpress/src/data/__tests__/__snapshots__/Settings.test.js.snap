// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Settings data access Get backup endpoint 1`] = `"graphql/v1/settings"`;

exports[`Test Settings data access Get endpoint 1`] = `"graphql/v1/settings"`;
